Add tools to help update of rules. (#22)
cybozu-go/setup-hw: docs/collector.md

Collect Redfish Data
====================

`collector` is a command-line tool that collects [Redfish][] data for creating/updating [collection rules](rule.md).

Synopsis
--------

```console
$ collector show [--input-file=<file>] [--base-rule=<file>] [--keys-only] [--omitempty] [--no-dup] [--ignore-field=<field>...] [--required-field=<field>...]
$ collector generate-rule [--input-file=<file>] [--base-rule=<file>] [--key=<key>:<type>...]
```

Description
-----------

`collector` is a command-line tool that collects and processes [Redfish][] data.
It shows the whole data, or generates a rule files.

In any mode, `collector` uses pre-collected Redfish data if `--input-file` is specified,
or collects Redfish data by accessing Redfish API if `--input-file` is not specified.

If `--base-rule` is specified, `collector` will exclude data included in `Traverse.Excludes`

> And path will be unified according to pattern specified in `Metrics.Path`.

Reviewer: ```suggestion
and paths are unified according to patterns specified in `Metrics.Path`.
```
